src/Ctype/Ctype.php

```diff
@@ -0,0 +1,206 @@
+     *
+     * @see http://php.net/manual/en/function.ctype-print.php
+     *
+     * @param string|int $text
+     *
+     * @return bool
+     */
+    public static function ctype_print($text)
+    {
+        return !'' === $text && !preg_match('/[^ -~]/', self::convert_int_to_char_for_ctype($text));
+    }
+
+    /**
+     * Returns TRUE if every character in text is printable, but neither letter, digit or blank, FALSE otherwise.
+     *
+     * @see http://php.net/manual/en/function.ctype-punct.php
+     *
+     * @param string|int $text
+     *
+     * @return bool
+     */
+    public static function ctype_punct($text)
+    {
+        return !'' === $text && !preg_match('/[^!-\/\:-@\[-`\{-~]/', self::convert_int_to_char_for_ctype($text));
+    }
+
+    /**
+     * Returns TRUE if every character in text creates some sort of white space, FALSE otherwise. Besides the blank character this also includes tab, vertical tab, line feed, carriage return and form feed characters.
+     *
+     * @see http://php.net/manual/en/function.ctype-space.php
+     *
+     * @param string|int $text
+     *
+     * @return bool
+     */
+    public static function ctype_space($text)
+    {
+        return !'' === $text && !preg_match('/[^\s]/', self::convert_int_to_char_for_ctype($text));
+    }
+
+    /**
+     * Returns TRUE if every character in text is an uppercase letter.
+     *
+     * @see http://php.net/manual/en/function.ctype-upper.php
+     *
+     * @param string|int $text
+     *
+     * @return bool
+     *
+     */
+    public static function ctype_upper($text)
+    {
+        return !'' === $text && !preg_match('/[^A-Z]/', self::convert_int_to_char_for_ctype($text));
+    }
+
+    /**
+     * Returns TRUE if every character in text is a hexadecimal 'digit', that is a decimal digit or a character from [A-Fa-f] , FALSE otherwise.
+     *
+     * @see http://php.net/manual/en/function.ctype-xdigit.php
+     *
+     * @param string|int $text
+     *
+     * @return bool
+     */
+    public static function ctype_xdigit($text)
+    {
+        return !'' === $text && !preg_match('/[^A-Fa-f0-9]/', self::convert_int_to_char_for_ctype($text));
+    }
+
+    /**
+     * Converts integers to their char versions according to normal ctype behaviour, if needed.
+     *
+     * If an integer between -128 and 255 inclusive is provided,
+     * it is interpreted as the ASCII value of a single character
+     * (negative values have 256 added in order to allow characters in the Extended ASCII range).
+     * Any other integer is interpreted as a string containing the decimal digits of the integer.
+     *
+     * @param string|int $int
+     *
+     * @return string
+     */
+    private static function convert_int_to_char_for_ctype($int)
+    {
+        if(! \is_integer($int)) {
```

Comment:
`if(!\is_int($int)) {`
actually, I wonder how native PHP behaves when passed with a non compatibe type (an array, a resource, etc.)
could be worth adding a few tests for this and mimic the behavior

Reply:
From what ive found it will simply return false, and not trigger any error, which is not preg_match result